eth/client.go

```diff
@@ -738,8 +739,113 @@ func (c *client) Vote(pollAddr ethcommon.Address, choiceID *big.Int) (*types.Tra
 	return poll.Vote(opts, choiceID)
 }
 
+func (c *client) Reward() (*types.Transaction, error) {
+	addr := c.accountManager.Account().Address
+
+	currentRound, err := c.CurrentRound()
+	if err != nil {
+		return nil, err
+	}
+
+	ep, err := c.GetTranscoderEarningsPoolForRound(c.accountManager.Account().Address, currentRound)
+	if err != nil {
+		return nil, err
+	}
+
+	mintable, err := c.CurrentMintableTokens()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get current mintable tokens")
+	}
+
+	totalBonded, err := c.GetTotalBonded()
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get total bonded")
+	}
+
+	// reward = (current mintable tokens for the round * active transcoder stake) / total active stake
+	reward := new(big.Int).Div(new(big.Int).Mul(mintable, ep.TotalStake), totalBonded)
+
+	hints, err := c.simulateTranscoderPoolUpdate(addr, reward.Add(reward, ep.TotalStake))
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to calculate hints, submitting reward transaction without hints")
+	}
+	return c.RewardWithHint(hints.PosPrev, hints.PosNext)
+}
+
 // Helpers
 
+// simulateTranscoderPoolUpdate simulates an update to the transcoder pool and returns the positional hints for a transcoder accordingly.
+// if the transcoder will not be in the updated set no hints will be returned
+func (c *client) simulateTranscoderPoolUpdate(del ethcommon.Address, newStake *big.Int) (lpTypes.StakingHints, error) {
+
+	// get max pool size
+	maxSize, err := c.GetTranscoderPoolMaxSize()
+	if err != nil {
+		return lpTypes.StakingHints{}, err
+	}
+	maxSizeInt := int(maxSize.Int64())
+
+	// get the transcoder pool
+	transcoders, err := c.TranscoderPool()
+	if err != nil {
+		return lpTypes.StakingHints{}, err
+	}
+
+	// if the transcoder pool is full and the newStake for 'del' is less than that of the last transcoder
```

Comment:
"less" should be "less than or equal to" since the transcoder will not be able to join the pool if its stake is equal to the stake of the last transcoder in the pool - see [this code](https://github.com/livepeer/protocol/blob/826af1124f645a819b54eb0500ddc31f0d2af755/contracts/bonding/BondingManager.sol#L1034).

Reply:
actually removed this line in the latest commit